src/test/java/com/google/cloud/genomics/dataflow/pipelines/VerifyBamIdTest.java

```diff
@@ -0,0 +1,322 @@
+import com.google.cloud.dataflow.sdk.options.PipelineOptionsFactory;
+import com.google.cloud.dataflow.sdk.testing.DataflowAssert;
+import com.google.cloud.dataflow.sdk.testing.TestPipeline;
+import com.google.cloud.dataflow.sdk.transforms.Create;
+import com.google.cloud.dataflow.sdk.transforms.DoFnTester;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.transforms.join.CoGbkResult;
+import com.google.cloud.dataflow.sdk.transforms.join.CoGroupByKey;
+import com.google.cloud.dataflow.sdk.transforms.join.KeyedPCollectionTuple;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.cloud.dataflow.sdk.values.TupleTag;
+import com.google.cloud.genomics.dataflow.coders.GenericJsonCoder;
+import com.google.cloud.genomics.dataflow.model.AlleleFreq;
+import com.google.cloud.genomics.dataflow.model.ReadBaseQuality;
+import com.google.cloud.genomics.dataflow.model.ReadCounts;
+import com.google.cloud.genomics.dataflow.model.ReadQualityCount;
+import com.google.cloud.genomics.dataflow.pipelines.VerifyBamId.FilterFreq;
+import com.google.cloud.genomics.dataflow.pipelines.VerifyBamId.GetAlleleFreq;
+import com.google.cloud.genomics.dataflow.pipelines.VerifyBamId.PileupAndJoinReads;
+import com.google.cloud.genomics.dataflow.pipelines.VerifyBamId.SampleReads;
+import com.google.cloud.genomics.dataflow.pipelines.VerifyBamId.SplitReads;
+import com.google.common.collect.ImmutableList;
+import com.google.genomics.v1.CigarUnit;
+import com.google.genomics.v1.CigarUnit.Operation;
+import com.google.genomics.v1.LinearAlignment;
+import com.google.genomics.v1.Read;
+import com.google.genomics.v1.Variant;
+import com.google.protobuf.ListValue;
+import com.google.protobuf.Value;
+
+import com.beust.jcommander.internal.Lists;
+
+import org.hamcrest.CoreMatchers;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests for the VerifyBamId pipeline.
+ */
+public class VerifyBamIdTest {
+  
+  @Test
+  public void testSplitReads() {
+    DoFnTester<Read, KV<Position, ReadBaseQuality>> splitReads = DoFnTester.of(new SplitReads());
+    
+    // single matched base -> one SingleReadQuality proto
+    Read r = Read.newBuilder()
+        .setProperPlacement(true)
+        .setAlignment(LinearAlignment.newBuilder()
+            .setPosition(com.google.genomics.v1.Position.newBuilder()
+                .setReferenceName("1")
+                .setPosition(123))
+            .addCigar(CigarUnit.newBuilder()
+                .setOperation(Operation.ALIGNMENT_MATCH)
+                .setOperationLength(1)))
+        .setAlignedSequence("A")
+        .addAlignedQuality(3)
+        .build();
+    Assert.assertThat(splitReads.processBatch(r), CoreMatchers.hasItems(KV.of(new Position()
+            .setReferenceName("1")
+            .setPosition(123L),
+            new ReadBaseQuality("A", 3))));
+    
+    // two matched bases -> two SingleReadQuality protos
```

Comment:
each of these tests are independent of the others in testSplitReads() right?  If so, it's useful to separate them out into separate test functions that are self descriptive, so on failure you don't have to dig into the test code to find out which particular case failed:

testSplitReads_singleMatchedBase()
testSplitReads_twoMatchedBases()

etc..